### PR TITLE
Logout functionality

### PR DESCRIPTION
**Pull request by glossopm**

### CODE REVIEW

File: routes/login/logout.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+logout_blueprint = Blueprint('logout', __name__)
+
+
+@logout_blueprint.route("/logout")
+def logout():
+    if is_active():
+        logout_user()
+        flash('Logout successful.')
+        return redirect(url_for('login.login')) 
+    else:
```
Comment (MikeSilverstone) [inline routes/login/logout.py]: there's no need to have the else here. you can just write the if and leave the rest outside. E.G:

```
if foo():
    bar

baz
```

File: tests/test_login.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 from unittest import mock
 from unittest.mock import MagicMock
 from globals.globals import bcrypt
+from flask import url_for
+from urllib.parse import urlparse
 
 class MockUser:
     def __init__(self, id, username, email, password):
```
Comment (MikeSilverstone) [inline tests/test_login.py]: What this constructor doing?

File: routes/login/logout.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from flask import render_template, Blueprint, redirect, url_for, flash
+from flask_login import logout_user, login_required, current_user
+from globals.globals import login_manager
+from db.models import User
```
Comment (MikeSilverstone) [inline routes/login/logout.py]: a lot of these are unused imports

File: routes/login/logout.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+def logout():
+    if is_active():
+        logout_user()
+        flash('Logout successful.')
+        return redirect(url_for('login.login')) 
+    else:
+        flash('No user currently logged in.')
+        return redirect(url_for('login.login'))
+
+def is_active():
```
Comment (MikeSilverstone) [inline routes/login/logout.py]: there should be two spaces between functions if they're not in a class

File: tests/test_login.py [hunk 5]
```diff
@@ -26,8 +28,13 @@ def test_login_page_response(self):
     @mock.patch("routes.login.login.login_user", MagicMock(return_value=False))
     def test_successful_login(self):  
         with self.client as client:
+<<<<<<< HEAD
```
Comment (MikeSilverstone) [inline tests/test_login.py]: this doesn't look right...

**Review by MikeSilverstone — CHANGES_REQUESTED** [review]

File: tests/test_login.py (hunk 2 above)
Comment (glossopm) [reply tests/test_login.py]: Used in mock.patch on line 26.

**Review by MikeSilverstone — APPROVED** [review]

**Pull request merged**